File: scala/src/main/scala/com/spoonacular/client/model/InlineResponse20027.scala
package com.spoonacular.client.model

import io.circe._
import io.finch.circe._
import io.circe.generic.semiauto._
import io.circe.java8.time._
import spoonacular._
import com.spoonacular.client.model.InlineResponse2007
import scala.collection.immutable.Seq

/**
 * 
 * @param products 
 * @param totalProducts 
 * @param _type 
 * @param offset 
 * @param number 
 */
case class InlineResponse20027(products: Seq[InlineResponse2007],
                totalProducts: Int,
                _type: String,
                offset: Int,
                number: Int
                )

object InlineResponse20027 {
    /**
     * Creates the codec for converting InlineResponse20027 from and to JSON.
     */
    implicit val decoder: Decoder[InlineResponse20027] = deriveDecoder
    implicit val encoder: ObjectEncoder[InlineResponse20027] = deriveEncoder
}

File: dart/com/spoonacular/client/model/inline_response20027.dart
part of openapi.api;

class InlineResponse20027 {
  
  List<InlineResponse2007> products = [];
  
  int totalProducts = null;
  
  String type = null;
  
  int offset = null;
  
  int number = null;
  InlineResponse20027();

  @override
  String toString() {
    return 'InlineResponse20027[products=$products, totalProducts=$totalProducts, type=$type, offset=$offset, number=$number, ]';
  }

  InlineResponse20027.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    if (json['products'] == null) {
      products = null;
    } else {
      products = InlineResponse2007.listFromJson(json['products']);
    }
    if (json['totalProducts'] == null) {
      totalProducts = null;
    } else {
          totalProducts = json['totalProducts'];
    }
    if (json['type'] == null) {
      type = null;
    } else {
          type = json['type'];
    }
    if (json['offset'] == null) {
      offset = null;
    } else {
          offset = json['offset'];
    }
    if (json['number'] == null) {
      number = null;
    } else {
          number = json['number'];
    }
  }

  Map<String, dynamic> toJson() {
    Map <String, dynamic> json = {};
    if (products != null)
      json['products'] = products;
    if (totalProducts != null)
      json['totalProducts'] = totalProducts;
    if (type != null)
      json['type'] = type;
    if (offset != null)
      json['offset'] = offset;
    if (number != null)
      json['number'] = number;
    return json;
  }

  static List<InlineResponse20027> listFromJson(List<dynamic> json) {
    return json == null ? new List<InlineResponse20027>() : json.map((value) => new InlineResponse20027.fromJson(value)).toList();
  }

  static Map<String, InlineResponse20027> mapFromJson(Map<String, dynamic> json) {
    var map = new Map<String, InlineResponse20027>();
    if (json != null && json.isNotEmpty) {
      json.forEach((String key, dynamic value) => map[key] = new InlineResponse20027.fromJson(value));
    }
    return map;
  }
}

